third_party/glad/BUILD.gn

```diff
@@ -0,0 +1,76 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+_glad_gen_dir = "$root_out_dir/gen/flutter/third_party/glad"
+
+config("glad_public") {
+  include_dirs = [ "$_glad_gen_dir/include" ]
+}
+
+source_set("glad") {
+  public_configs = [ ":glad_public" ]
+
+  public = [
+    "$_glad_gen_dir/include/KHR/khrplatform.h",
+    "$_glad_gen_dir/include/glad/glad.h",
+  ]
+
+  sources = [
+    "$_glad_gen_dir/src/glad.c",
+  ]
+
+  deps = [
+    ":gen_glad_sources",
+  ]
+}
+
+action("gen_glad_sources") {
+  script = "$flutter_root/third_party/glad/generate.py"
+  inputs = [
+    "generate.py",
+    "scripts/glad/__init__.py",
+    "scripts/glad/__main__.py",
+    "scripts/glad/opener.py",
+    "scripts/glad/parse.py",
+    "scripts/glad/spec.py",
+    "scripts/glad/util.py",
+    "scripts/glad/files/__init__.py",
+    "scripts/glad/files/egl.xml",
+    "scripts/glad/files/glx.xml",
+    "scripts/glad/files/wgl.xml",
+    "scripts/glad/files/gl.xml",
+    "scripts/glad/files/khrplatform.h",
+    "scripts/glad/lang/__init__.py",
+    "scripts/glad/lang/common/__init__.py",
+    "scripts/glad/lang/common/generator.py",
+    "scripts/glad/lang/common/loader.py",
+    "scripts/glad/lang/common/util.py",
+    "scripts/glad/lang/c/__init__.py",
+    "scripts/glad/lang/c/debug.py",
+    "scripts/glad/lang/c/generator.py",
+    "scripts/glad/lang/c/loader/__init__.py",
+    "scripts/glad/lang/c/loader/gl.py",
+    "scripts/glad/lang/c/loader/wgl.py",
+    "scripts/glad/lang/c/loader/egl.py",
+    "scripts/glad/lang/c/loader/glx.py",
+  ]
+
+  outputs = [
+    "$_glad_gen_dir/src/glad.c",
+    "$_glad_gen_dir/include/glad/glad.h",
+    "$_glad_gen_dir/include/KHR/khrplatform.h",
```

Comment:
Rather than duplicate these, please make file-local variables, one for the list of generated sources and one for the list of generated headers, and use those above and then again here (added together).